Dockerfile

```
@@ -10,13 +10,22 @@ WORKDIR /opt/app/
 # System deps and get the ohsome-api repository data:
 RUN apk add git bash maven && git clone https://github.com/GIScience/ohsome-api.git ./ && git fetch --all --tags
```

Why don't you clone the branch/tag directly. You don't need all the other branches.
```suggestion
RUN apk add git bash maven \
     && if [ -z $OHSOMEAPI_VERSION ] || [ "$OHSOMEAPI_VERSION" = "latest" ] ; then echo Version not provided. Sticking to latest version. && export BRANCH_PARAMETER=""; else echo Version provided. Checkout $OHSOMEAPI_VERSION && export BRANCH_PARAMETER="-b $OHSOMEAPI_VERSION"; fi  \
    && git clone https://github.com/GIScience/ohsome-api.git $BRANCH_PARAMETER --depth 1 .
```

---

Fancy. In general we don't need to over engineer the build step since these layers are not going into the final image.